### PR TITLE
version name as __version__

### PR DESCRIPTION
**Pull request by weikang9009**
* Change `version` to ` __version__`
* Delete version.py in example folder 
* Update README.rst - DELETE instruction on `import libpysal.api` since there is not an `api.py`

### CODE REVIEW

**Comment by ljwolf** [comment]
This will still provide `version` and `__version__` under `libpysal`. Reading around, I see a few common patterns in our peer packages

1. define `__version__` directly in `__init__.py` (seaborn)
2. use versioneer (geopandas)
3. write the version info at install, from within setup.py (networkx, statsmodels). 

I would prefer 1, and have setup import from that? We also don't need to define `stable release date` anymore.

**Comment by weikang9009** [comment]
Right.  The current implementation requires defining version twice in two different places.

I also found that`Numpy` and `Scipy` adopt the third approach and define `__version__` in `setup.py`.

`Seaborn` adopts the first approach, but still define version twice: one in "setup.py" and the other in `__ini__.py`. Your suggestion of importing `__version__` from `__ini__.py` so that we only have to define it once sounds like a great enhancement.

I am fine with option 1 (with your adjustment) and option 3.

**Comment by weikang9009** [comment]
See [this](https://milkr.io/kfei/5-common-patterns-to-version-your-Python-package/1) for an interesting discussion on versioning a Python package. 

It is suggested that option 1 (with your adjustment) **will fail if you import dependencies in  `__init__.py` which are not installed on the user's environment, which is very likely to happen.** Keeping `version.py` will not result in such problem.

**Comment by ljwolf** [comment]
how would you feel about about writing `__version__` on the first line and using:
```python
with open('libpysal/__init__.py') as f:
    version = f.readline().split('=')[1].strip()
```

**Comment by ljwolf** [comment]
I should clarify why I think that's a good option :smile: 

it's [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), with the version being in one place. 
It avoids importing the module to set it up.

**Comment by weikang9009** [comment]
Looks good! So it will bypass the import issue. But we will have to define`__version__` in the first line of `__ini__.py`.

File: setup.py [hunk 1]
```diff
@@ -9,12 +9,10 @@
...
 
-MAJOR = 3
-MINOR = 0
-MICRO = 7
-ISRELEASED = False
-VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
-
+# Get __version__ from libpysal/__init__.py without importing the package
+# __version__ has to be defined in the firt line
+exec(compile(open('libpysal/__init__.py').readline(),
```
[the remark this answers is not in the document]
Comment (weikang9009) [reply setup.py]: You are right!

**Pull request merged**